vms/subnet.go

```diff
@@ -0,0 +1,345 @@
+	if err := finalizeBlockchain(ctx, args.log, args.allNodes, blockchainID); err != nil {
+		return fmt.Errorf("error checking all nodes are validating subnet: %w", err)
+	}
+
+	return nil
+}
+
+// initialize shared args
+func newArgs(
+	log logging.Logger,
+	vm CustomVM,
+	network network.Network,
+	fundedPChainPrivateKey string,
+) (*args, error) {
+	userPass := defaultUserPass
+
+	allNodes, err := network.GetAllNodes()
+	if err != nil {
+		return nil, err
+	}
+
+	txNodeNames, err := network.GetNodeNames()
+	if err != nil {
+		return nil, err
+	}
+
+	if len(txNodeNames) == 0 {
+		return nil, errors.New("the array of node names is empty! Can't get any nodes")
+	}
+	// txNode will be the node we issue all transactions on
+	txNode := allNodes[txNodeNames[0]]
+	txClient := txNode.GetAPIClient()
+	// we need to create a user for platformvm calls
+	ok, err := txClient.KeystoreAPI().CreateUser(userPass)
+	if !ok || err != nil {
+		return nil, fmt.Errorf("could not create user: %w", err)
+	}
+
+	txPChainClient := txClient.PChainAPI()
+	// Import genesis key
+	fundedAddress, err := txPChainClient.ImportKey(userPass, fundedPChainPrivateKey)
+	if err != nil {
+		return nil, fmt.Errorf("unable to import genesis key: %w", err)
+	}
+
+	rSubnetID, err := ids.FromString(vm.SubnetID)
+	if err != nil {
+		return nil, fmt.Errorf("invalid subnetID string: %w", err)
+	}
+	return &args{
+		log:            log,
+		txPChainClient: txPChainClient,
+		fundedAddress:  fundedAddress,
+		userPass:       userPass,
+		allNodes:       allNodes,
+		rSubnetID:      rSubnetID,
+	}, nil
+}
+
+// createSubnet issues the CreateSubnet transaction and waits for
+// it to be accepted. It returns an error if the transaction failed
+// or there was a timout.
+func createSubnet(ctx context.Context, args *args) error {
+	// Create a subnet
+	subnetIDTx, err := args.txPChainClient.CreateSubnet(
+		args.userPass,
+		[]string{args.fundedAddress},
+		args.fundedAddress,
+		[]string{args.fundedAddress},
+		defaultKeyThreshold,
+	)
+	if err != nil {
+		return fmt.Errorf("unable to create subnet: %w", err)
+	}
+
+	// wait until all nodes have accepted the CreateSubnet transaction
+	return utils.AwaitedAllNodesPChainTxAccepted(
+		ctx,
+		args.log,
+		apiRetryFreq,
+		args.allNodes,
+		subnetIDTx)
+}
+
+// isSubnetInList returns an error if the given subnet is not in the client's list
+func isSubnetInList(client platformvm.Client, rSubnetID ids.ID) error {
+	// confirm created subnet appears in subnet list
+	var subnetAPIs []platformvm.APISubnet
+	var err error
+	if subnetAPIs, err = client.GetSubnets([]ids.ID{rSubnetID}); err != nil {
+		return fmt.Errorf("subnet not found: %w", err)
+	}
+	// we should assume that the array returned has length 0 because it is filtered;
+	// but because it's an array let's be thorough
+	found := false
+	for _, api := range subnetAPIs {
+		if api.ID == rSubnetID {
+			found = true
+		}
+	}
+	if !found {
+		return errors.New("subnet not in returned list")
+	}
```

Comment:
Sorry, I was wrong before -- we don't need to check that the subnet ID is in the returned list since we're filtering by the subnet ID already. We just have to assert that the length of the returned list is 1